### PR TITLE
Detection of surface to clean

### PR DESCRIPTION
**Pull request by Kebanee**
It know stops instead of driving around the area. Specific surfaces/points to clean will be implemented.

### CODE REVIEW

File: smartcar.ino [hunk 1]
```diff
@@ -0,0 +1,138 @@
+#include <vector>
```
Comment (platisd) [inline smartcar.ino:1]: Why is this file here?

**Review by platisd — DISMISSED** [review]

File: arduino/smartcar/smartcar.ino [hunk 2]
```diff
@@ -136,3 +136,16 @@ void preventCrashWithUSandIR(DistanceSensor& distanceSensor) {
...
+  if (distance > minDistance && distance < obstacleAvoidDistance) {
+      Serial.print(distanceSensor.getDistance());
+      Serial.println(" Obstacle detected. ");
+      car.setSpeed(0);
+      delay(20000);
+      car.overrideMotorSpeed(50, -100);
+      delay (800);
+      car.setSpeed(0);
+      delay (20000);
+  }
```
Comment (platisd) [inline arduino/smartcar/smartcar.ino:150]: What's using this and how is this significantly different from `preventCrashWithUSandIR`?

Comment (Kebanee) [reply arduino/smartcar/smartcar.ino:150]: This was just a small adjustment and is not completely finished. I figured since you and Krasen prefered that we would make some smaller contributions for you to keep track of some work. The rest of this task will be finished today.

**Comment by Kebanee** [comment]
So I and Graciela worked on this now and it does basic cleaning of areas but we are going to implement a "wall following task"

File: arduino/smartcar/smartcar.ino (hunk 2 above)
Comment (platisd) [reply arduino/smartcar/smartcar.ino]: > This was just a small adjustment and is not completely finished. I figured since you and Krasen prefered that we would make some smaller contributions for you to keep track of some work. The rest of this task will be finished today.

Just to clarify, we want smaller **functional** pieces of work. If something is not used it is by definition not functional.

File: smartcar.ino (hunk 1 above)
Comment (platisd) [reply smartcar.ino:1]: This file is... still here! :scream_cat:

Comment (alynnka) [reply smartcar.ino:1]: File removed

Comment (platisd) [reply smartcar.ino:1]: It's still here 😱

File: smartcar/smartcar.ino [hunk 3]
```diff
@@ -0,0 +1,219 @@
+#include <MQTT.h>
```
Comment (platisd) [inline smartcar/smartcar.ino:1]: I don't think this file should be here either!

Comment (alynnka) [reply smartcar/smartcar.ino:1]: File #include <vector> removed. 
Sketch does not compile without fiile #include <MQTT.h>

Comment (platisd) [reply smartcar/smartcar.ino:1]: The entire _file_ I commented on should be removed, it's a duplicate

**Comment by alynnka** [comment]
Removed file duplicated

File: smartcar.ino (hunk 1 above)
Comment (alynnka) [reply smartcar.ino]: Removed file duplicated

**Review by alynnka — APPROVED** [review]
Extra files deleted
